src/chk/chk_engine.c

```diff
@@ -669,28 +734,46 @@ chk_engine_start(uint64_t gen, uint32_t rank_nr, d_rank_t *ranks,
 	D_ASSERT(rank_list != NULL);
 	D_ASSERT(d_list_empty(&ins->ci_pool_list));
 	D_ASSERT(d_list_empty(&ins->ci_pending_list));
-	D_ASSERT(ins->ci_sched == ABT_THREAD_NULL);
 
-	if (ins->ci_iv_ns != NULL) {
-		ds_iv_ns_put(ins->ci_iv_ns);
-		ins->ci_iv_ns = NULL;
-	}
+	if (ins->ci_sched != ABT_THREAD_NULL)
+		ABT_thread_free(&ins->ci_sched);
 
-	if (ins->ci_iv_group != NULL) {
-		crt_group_secondary_destroy(ins->ci_iv_group);
-		ins->ci_iv_group = NULL;
+	chk_iv_ns_cleanup(&ins->ci_iv_ns);
```

Comment:
this is a bit strange in chk_engine_start().  chk_engine_stop() won't be called in some cases?

Reply:
The engine scheduler is not started yet under such case.